### PR TITLE
Bug 1832241: Don't set net-attach-def type at root

### PR DESCRIPTION
**Pull request by yaacov**
Continuing #5290

Screenshot:
Peek 2020-05-13 12-30

### CODE REVIEW

**Comment by yaacov** [comment]
@phoracek @pcbailey please review

File: frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx [hunk 1]
```diff
@@ -27,11 +27,17 @@ import { networkTypeParams, networkTypes } from '../../constants';
 import NetworkTypeOptions from './NetworkTypeOptions';
 
 const buildConfig = (name, networkType, typeParamsData): NetworkAttachmentDefinitionConfig => {
-  const config: NetworkAttachmentDefinitionConfig = {
-    name,
-    type: networkType,
-    cniVersion: '0.3.1',
-  };
+  const config: NetworkAttachmentDefinitionConfig =
+    networkType === 'cnv-bridge'
```
Comment (phoracek) [inline frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx:31]: If we need this `if` here and on line 49, maybe it would make sense to just combine them? The current state of the code is a little hard to follow. And if we extend it with additional types, it may bring a lot of `if`s.

Just a suggestion, won't fight for that

Comment (yaacov) [reply frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx]: :+1: fixed

File: frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx [hunk 2]
```diff
@@ -41,6 +41,9 @@ const buildConfig = (name, networkType, typeParamsData): NetworkAttachmentDefini
   }
 
   if (networkType === 'cnv-bridge') {
+    // cnv-bridge should not define type on root
+    delete config.type;
```
Comment (pcbailey) [inline frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionsForm.tsx:45]: This is a nice way to handle this. We can always change it if we find that others tend not to define the type at the root, as well.
